src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -2134,28 +2137,30 @@ private bool SkipMultiLineComment(ReadOnlySpan<byte> localBuffer, out int idx)
                 }
                 if (foundIdx != 0 && localBuffer[foundIdx + idx - 1] == JsonConstants.Asterisk)
                 {
-                    idx += foundIdx;
+                    // foundIdx points to '*' in the end-of-comment delimiter. Hence increment idx by one position
+                    // less to make it point right before beginning of end-of-comment delimiter i.e. */
+                    idx += foundIdx - 1;
                     break;
                 }
                 idx += foundIdx + 1;
             }
 
-            Debug.Assert(idx >= 1);
-
             // Consume the /* and */ characters that are part of the multi-line comment.
-            // Since idx is pointing at right after the final '*' (i.e. before the last '/'), we don't need to count that character.
-            // Hence, we increment consumed by 3 (instead of 4).
-            _consumed += 4 + idx - 1;
+            // idx points right before the final '*' (i.e. before the last '/'). Hence increment _consumed
```

Comment:
I interpreted the clause `(i.e. before the last '/')` in the original comment as description of the `*` i.e. _the '*' which is before last '/'_ e.g. it could be `**/` towards the end. Hence I retained `(i.e. ... '/')`.

Hmm. So IIUC, you mean I should limit the comment to `// idx points right before the final '*'. Hence increment _consumed`... right?

Reply:
Oh, you are right. It's fine to keep it as is. Maybe we can adjust it to be as follows:

```suggestion
            // idx points right before the final '*' (which is right before the last '/'). Hence increment _consumed...
```

Reply:
Done!